server/src/main/java/org/elasticsearch/common/file/AbstractFileWatchingService.java

```diff
@@ -385,5 +393,5 @@ long retryDelayMillis(int failedCount) {
      * Holds information about the last known state of the file we watched. We use this
      * class to determine if a file has been changed.
      */
-    record FileUpdateState(long timestamp, String path, Object fileKey) {}
+    public record FileUpdateState(long timestamp, String path, Object fileKey) {}
```

Comment:
This is an implementation detail of the watching service, so I do not think it should be public. It seems like it is only used in this file?

Reply:
Good call -- I missed this because the code gets this reference via the subclass. We can make it private.